jscomp/gentype_tests/typescript-react-example/src/Comments.gen.tsx

```diff
@@ -8,23 +8,27 @@ const CommentsBS: any = CommentsBS__Es6Import;
 
 // tslint:disable-next-line:interface-over-type-literal
 export type DecideSubject_payload = { 
-/** A hint to use as a guide when thinking of your poem.*/
+/** A hint to use as a guide when thinking of your poem. */
 readonly hint: string };
 
 // tslint:disable-next-line:max-classes-per-file 
+/** The input used to generate the prompt and system prompt. */
 export abstract class DecideSubject_input { protected opaque!: any }; /* simulate opaque types */
 
 // tslint:disable-next-line:interface-over-type-literal
+/** The output from evaluating the llm prompt */
 export type DecideSubject_output = {
-  /** The text of the poem.*/
+  /** The text of the poem. */
   readonly text: string; 
-  /** The prompt used to generate the poem.*/
+  /** The prompt used to generate the poem. */
   readonly prompt: string; 
-  /** The system prompt used to generate the poem.*/
+  /** The system prompt used to generate the poem. */
   readonly systemPrompt: string
 };
 
-/**  Decide on a subject matter for a poem. */
+/** Decide on a subject matter for a poem. */
 export const DecideSubject__placeholder: (run:string, times:number) => void = CommentsBS.DecideSubject._placeholder;
 
-export const DecideSubject: { _placeholder: (run:string, times:number) => void } = CommentsBS.DecideSubject
+export const DecideSubject: { 
```

Comment:
This `const` represents the `DecideSubject` module in `Comments.res`, and it has a comment in the original res file, but I cannot for the life of me figure out how to propagate that comment out to here. It would be very valuable though, comments for modules are important.

@cristianoc any ideas?

Reply:
gentype does not store info about modules.
It stores info about values, and what module they belong to.
I guess this would go to some separate module info somewhere.

Reply:
After trying several different approaches with no luck I conclude this seem to require some fairly deep changes. Maybe I'm missing something valuable.

The rest in this PR is useful by itself - maybe we could review and merge what's in this PR, and then do module comments separately.